discovery-infra/test_infra/helper_classes/entity_operations.py

```diff
@@ -0,0 +1,61 @@
+from abc import ABC, abstractmethod
+
+from test_infra.helper_classes.config import BaseEntityConfig, BaseClusterConfig, BaseInfraEnvConfig
+
+
+class _EntityOperations(ABC):
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        raise NotImplemented
+
+    @property
+    @abstractmethod
+    def terraform_params(self) -> dict:
+        raise NotImplemented
+
+    @property
+    @abstractmethod
+    def name_suffix(self) -> str:
+        raise NotImplemented
+
+
+class _ClusterOperations(_EntityOperations):
+    def __init__(self, controller, entity_config: BaseClusterConfig):
+        self._controller = controller
+        self._entity_config = entity_config
+
+    @property
+    def name(self) -> str:
+        return self._entity_config.cluster_name.get()
+
+    @property
+    def name_suffix(self) -> str:
+        return self._entity_config.cluster_name.suffix
+
+    @property
+    def terraform_params(self) -> dict:
+        return {
+            "cluster_name": self.name,
+            "cluster_domain": self._entity_config.base_dns_domain
+        }
+
+
+class _InfraEnvOperations(_EntityOperations):
+    def __init__(self, controller, entity_config: BaseInfraEnvConfig):
+        self._controller = controller
+        self._entity_config = entity_config
+
+    @property
+    def name(self) -> str:
+        return self._entity_config.infra_env_name.get()
+
+    @property
+    def name_suffix(self) -> str:
+        return self._entity_config.infra_env_name.suffix
+
+    @property
+    def terraform_params(self) -> dict:
+        return {
+            "infra_env_name": self.name
+        }
```

Comment:
Can it be part of the configs? Why do we need another class?

Reply:
agree, sounds like much better. We have so many classes that i already got lost who is who

Reply:
Not sure.. I'll see if it can be changed

Reply:
I deleted this file